src/components/core/Modal.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from "react";
+
+const Modal = ({ item, onClose, onSave, mode = "edit" }) => {
+  const initialFormData = mode === "add" ? { title: "", category: "", price: "", rating: "", stock: "" } : { ...item };
+  const [formData, setFormData] = useState(initialFormData);
+
+  const handleChange = (key, value) => {
+    setFormData({ ...formData, [key]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (mode === "add") {
+      onSave(formData);
+    } else {
+      onSave({ ...formData, id: item.id });
+    }
+  };
+
+  return (
+    <div className="fixed inset-0 bg-gray-800 bg-opacity-75 flex items-center justify-center">
+      <div className="bg-white rounded-lg shadow-lg w-1/3 p-6">
+        <h2 className="text-xl font-bold mb-4">{mode === "add" ? "Add New Item" : "Edit Item"}</h2>
+        <form onSubmit={handleSubmit}>
+          {Object.keys(formData).map((key) => (
+            <div key={key} className="mb-4">
+              <label className="block text-sm font-medium mb-1">{key.charAt(0).toUpperCase() + key.slice(1)}</label>
+              <input
+                type="text"
+                value={formData[key]}
+                onChange={(e) => handleChange(key, e.target.value)}
+                className="w-full border px-3 py-2 rounded-lg"
+              />
+            </div>
+          ))}
+          <div className="flex justify-end">
+            <button
+              type="button"
+              className="bg-gray-500 text-white px-4 py-2 rounded mr-2"
+              onClick={onClose}
+            >
+              Cancel
+            </button>
+            <button
+              type="submit"
+              className="bg-blue-500 text-white px-4 py-2 rounded"
+            >
+              {mode === "add" ? "Add" : "Save"}
```

Comment:
use constants for "add" and "save"

Reply:
Created a file to store as constants.

Resolved✅